src/opensles/AudioOutputStreamOpenSLES.cpp

```diff
@@ -298,15 +298,23 @@ Result AudioOutputStreamOpenSLES::requestStart() {
     setDataCallbackEnabled(true);
 
     setState(StreamState::Starting);
+
+    if (getBufferDepth(mSimpleBufferQueueInterface) == 0) {
+        // Enqueue the first buffer if needed to start the streaming.
+        // We may need to stop the current stream.
+        bool shouldStopStream = processBufferCallback(mSimpleBufferQueueInterface);
+        if (shouldStopStream) {
+            LOGD("Stopping the current stream.");
+            if (requestStop_l() != Result::OK) {
+                LOGW("Failed to flush the stream. Error %s", convertToText(flush()));
+            }
+        }
+    }
+
     Result result = setPlayState_l(SL_PLAYSTATE_PLAYING);
```

Comment:
Can skip setting PLAYSTATE if shouldStopStream is true.

Reply:
Not needed as we are caching getBufferDepth